packages/lesswrong/server/collections/advisorRequests/mutations.ts

```diff
@@ -0,0 +1,150 @@
+      data,
+      schema,
+      skipValidation,
+    });
+
+    data = callbackProps.document;
+
+    data = await runFieldOnCreateCallbacks(schema, data, callbackProps);
+
+    const afterCreateProperties = await insertAndReturnCreateAfterProps(data, 'AdvisorRequests', callbackProps);
+    let documentWithId = afterCreateProperties.document;
+
+    await runCountOfReferenceCallbacks({
+      collectionName: 'AdvisorRequests',
+      newDocument: documentWithId,
+      callbackStage: 'createAfter',
+      afterCreateProperties,
+    });
+
+    return documentWithId;
+  },
+
+  updateFunction: async ({ selector, data }: UpdateAdvisorRequestInput, context, skipValidation?: boolean) => {
+    const { currentUser, AdvisorRequests } = context;
+
+    // Save the original mutation (before callbacks add more changes to it) for
+    // logging in FieldChanges
+    const origData = cloneDeep(data);
+
+    const {
+      documentSelector: advisorrequestSelector,
+      previewDocument, 
+      updateCallbackProperties,
+    } = await checkUpdatePermissionsAndReturnProps('AdvisorRequests', { selector, context, data, schema, skipValidation });
+
+    const { oldDocument } = updateCallbackProperties;
+
+    const dataAsModifier = dataToModifier(clone(data));
+    data = await runFieldOnUpdateCallbacks(schema, data, dataAsModifier, updateCallbackProperties);
+
+    let modifier = dataToModifier(data);
+
+    // This cast technically isn't safe but it's implicitly been there since the original updateMutator logic
+    // The only difference could be in the case where there's no update (due to an empty modifier) and
+    // we're left with the previewDocument, which could have EditableFieldInsertion values for its editable fields
+    let updatedDocument = await updateAndReturnDocument(modifier, AdvisorRequests, advisorrequestSelector, context) ?? previewDocument as DbAdvisorRequest;
+
+    await runCountOfReferenceCallbacks({
+      collectionName: 'AdvisorRequests',
+      newDocument: updatedDocument,
+      callbackStage: "updateAfter",
+      updateAfterProperties: updateCallbackProperties,
+    });
+
+    void logFieldChanges({ currentUser, collection: AdvisorRequests, oldDocument, data: origData });
+
+    return updatedDocument;
+  },
+});
+
+const wrappedCreateFunction = makeGqlCreateMutation(createFunction, {
+  newCheck,
+  accessFilter: (rawResult, context) => accessFilterSingle(context.currentUser, 'AdvisorRequests', rawResult, context)
+});
+
+const wrappedUpdateFunction = makeGqlUpdateMutation('AdvisorRequests', updateFunction, {
+  editCheck,
+  accessFilter: (rawResult, context) => accessFilterSingle(context.currentUser, 'AdvisorRequests', rawResult, context)
+});
+
+
+export { createFunction as createAdvisorRequest, updateFunction as updateAdvisorRequest };
+export { wrappedCreateFunction as createAdvisorRequestMutation, wrappedUpdateFunction as updateAdvisorRequestMutation };
+
+export const graphqlAdvisorRequestTypeDefs = gql`
```

Comment:
This is less-bad than how Vulcan did this (`initGraphQL` had some pretty extreme spaghetti), but there's still no reason for the copy-paste here. It's too dynamic for this to be benefitting from auto-extraction from source, so you might as well just make it a function.